ClassTrigger.trigger.txt

```diff
@@ -0,0 +1,23 @@
+trigger ClassTrigger on Class__c (before insert, before delete, before update){
+    
+    // trigger code to restrict class deletion if class has more than one female students
+    if(trigger.isDelete){
+        List<AggregateResult> students = [SELECT Class__r.id, COUNT(Name) FROM Student__c WHERE Sex__c = 'female' AND Class__c IN:  Trigger.old GROUP BY Class__r.id HAVING COUNT(Name) > 1];
+        for (AggregateResult everyStudent : students){
+            Trigger.oldMap.get((id)everyStudent.get('Class__r.id')).addError('This class cannot be deleted as it contains more than one female students');    
+        }
+    }
+    
+    // trigger code to delete all related students record if class custom status is changed to reset
+    if(trigger.isUpdate){
+        List<Id> classIds = new List<Id>();
+        for(Class__c cls : trigger.new){
+            Class__c oldRecord = trigger.oldMap.get(cls.Id);
+            if(oldRecord.Custom_Status__c != 'reset' && cls.Custom_Status__c == 'reset'){

```

Comment:
I would suggest that you use old list and new list to identify value changes, the approach you have used here would also work fine but it's just a more widely followed practice